client/src/c_console.cpp

```diff
@@ -352,12 +355,39 @@ void ConsoleCommandLine::moveCursorRight()
 
 void ConsoleCommandLine::moveCursorLeftWord()
 {
-	if (cursor_position > 0)
-		cursor_position--;
+	bool firstSpacesCleared = false;
+	bool spaceWord = false;
 
 	const char* str = text.c_str();
-	while (cursor_position > 0 && str[cursor_position - 1] != ' ')
+
+	if (str[cursor_position - 1] == ' ')
```

Comment:
What happens if `cursor_position` is 0?

Also, I'm not super sure it matters for the console, but in general checking for whitespace is doing with `isspace` or some other white-space detection function instead of hadrcoding 0x20.

Reply:
Changed to use `isspace()` and check `cursor_position` for all of these.